shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -239,16 +242,23 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
             SetupAppend setup = (SetupAppend) msg;
             setupSegments.put(new SimpleImmutableEntry<>(setup.getSegment(), setup.getWriterId()),
                               new Session(setup.getWriterId(), setup.getRequestId()));
+            flushRequired();
         } else if (msg instanceof BlockTimeout) {
             BlockTimeout timeoutMsg = (BlockTimeout) msg;
             if (tokenCounter.get() == timeoutMsg.token) {
                 breakCurrentAppend(out);
                 flushAll(out);
             }
+            flushRequired();
+        } else if (msg instanceof Hello) {
+            Preconditions.checkState(isChannelFree());
+            Preconditions.checkState(pendingWrites.isEmpty());
```

Comment:
This precondition will fail in a connection pooling case, right?

Reply:
The message is sent in `FlowHandler.channelRegistered()` is that invoked more than once on a pooled connection?

Reply:
no, it is invoked only once.